src/lib/device/journal_vdev.hpp

```diff
@@ -19,250 +19,367 @@
 #include <atomic>
 #include <functional>
 #include <memory>
+#include <vector>
+#include <condition_variable>
 
 #include "device.h"
+#include "physical_dev.hpp"
 #include "virtual_dev.hpp"
 
 namespace homestore {
 typedef std::function< void(const off_t ret_off) > alloc_next_blk_cb_t;
+using journal_id_t = uint64_t;
+// Each logstore family is associated to a single logdevice.
+using logdev_id_t = uint64_t;
+
+// Chunks used for journal vdev has journal related info stored in chunk private data.
+// Each log device has a list of journal chunk data with next_chunk.
+// Journal vdev will arrange the chunks in order during recovery.
+struct JournalChunkPrivate {
+    logdev_id_t logdev_id{0};
+    bool is_head{false};           // Is it the head element.
+    uint64_t created_at{0};        // Creation timestamp
+    uint64_t end_of_chunk_size{0}; // The offset indicates end of chunk.
```

Comment:
nit: I know the this is migrated from old, can we rename it to 'end_offset', since it is actually a offset not a size.

Reply:
Made it end_of_chunk

Reply:
Made it end_of_chunk